packages/ui-causal-graph-editor/src/graph-viewer/causal-graph-editor.tsx

```diff
@@ -120,6 +120,8 @@ function CausalGraphEditor(props: CausalGraphEditorProps): JSX.Element {
         props.graphLayout.requiresPosition
     );
 
+    const editorMode = props.editorMode ?? (isDag(state.graph) ? EditorMode.DEFAULT : EditorMode.PAG_VIEWER);
```

Comment:
This will re-calculate on each graph editor re-render. Is that what we really want? I would assume we want to infer it once when we first parse the graph?

Reply:
Changed to use useMemo instead. Thinking that if the variable is updated the outside and a DAG graph becomes not that, then in that case editormode would need to update

Reply:
Have you checked how often this runs? i.e. does it happen on every graph change, both internal and external? Just making sure we're not introducing something that runs too often

Reply:
Note: resolved off PR
This was not working as I thought, it would only run on first render rather than on graphData change. This behaviour of only doing it once however is probably better for the user. For it to be neater it was changed to use `useState` instead of `useMemo`